api/query/search.go

```diff
@@ -167,7 +167,7 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		logger := shared.GetLogger(ctx)
 		logger.Infof("Forwarding structured search request to %s: %s", hostname, string(data))
 
-		client := sh.api.GetHTTPClient()
+		client, _ := sh.api.GetSlowHTTPClient(time.Second * 15)
```

Comment:
```suggestion
		client, cancel := sh.api.GetSlowHTTPClient(time.Second * 15)
		defer cancel()
```

I think this would be a resource leak otherwise. (Tests might need to be fixed accordingly.)